ui/src/app/pages/data/data-set/dataset-page.tsx

```diff
@@ -519,60 +530,73 @@ const DataSetPage = fp.flow(withCurrentWorkspace(), withUrlParams())(
     }
 
     async getPreviewList() {
-      this.setState({previewList: [], previewDataLoading: true});
       const {namespace, id} = this.props.workspace;
-      const request = {
-        name: '',
-        description: '',
-        conceptSetIds: this.state.selectedConceptSetIds,
-        includesAllParticipants: this.state.includesAllParticipants,
-        cohortIds: this.state.selectedCohortIds,
-        prePackagedConceptSet: this.getPrePackagedConceptSet(),
-        values: this.state.selectedValues
-      };
-      try {
-        const dataSetPreviewResp = await dataSetApi().previewQuery(namespace, id, request);
-        this.setState({
-          previewList: dataSetPreviewResp.domainValue,
-          selectedPreviewDomain: dataSetPreviewResp.domainValue[0].domain
-        });
-      } catch (ex) {
-        const exceptionResponse = await ex.json() as unknown as ErrorResponse;
-        let errorText: string;
-        switch (exceptionResponse.statusCode) {
-          case 400:
-            if (exceptionResponse.message ===
-              'Data Sets must include at least one cohort and concept.') {
-              errorText = exceptionResponse.message;
-            } else if (exceptionResponse.message ===
-              'Concept Sets must contain at least one concept') {
-              errorText = 'One or more of your concept sets has no concepts. ' +
-                'Please check your concept sets to ensure all concept sets have concepts.';
-            }
-            break;
-          case 404:
-            if (exceptionResponse.message.startsWith(
-              'Not Found: No Cohort definition matching cohortId')) {
-              errorText = 'Error with one or more cohorts in the data set. ' +
-                'Please submit a bug using the contact support button';
-            }
-            break;
-          case 504:
-            if (exceptionResponse.message ===
-              'Timeout while querying the CDR to pull preview information.') {
-              errorText = 'Query to load data from the All of Us Database timed out. ' +
-                'Please either try again or export data set to a notebook to try there';
-            }
-            break;
-          default:
-            errorText = 'An unexpected error has occurred. ' +
-              'Please submit a bug using the contact support button';
+      const domains = fp.uniq(this.state.selectedValues.map(domainValue => domainValue.domain));
+      this.setState({
+        previewList: domains.map(domain => {
```

Comment:
I don't know if the requirements may change in the future where multiple domains would make sense but it seems like a map is the more natural data structure for the current access patterns

Reply:
The nice thing about a list is that it is ordered. I don't know if we care about ordering here, but that could be a valuable thing to maintain. Otherwise I agree with you 100%. Let's chat faces?

(Also, for what it's worth, we do support multiple domains. They are just loaded individually from the API as opposed to all at once)